### DISCUSSION TITLE
Port forwarding for specific Kind IP

### DISCUSSION THREAD

**Discussion by wackazong** [lima-vm/lima#2087] [Show and tell]
I set up a Kind Kubernetes Cluster in Colima according to https://opencredo.com/blogs/building-the-best-kubernetes-test-cluster-on-macos/

I want to use port forwarding to forward the local host ports 80 and 443 to a specific IP in the Kind network. How can I do that?

Goal is to be able to use the `.localhost` domain in Chrome for development.

**Comment by afbjorklund** [comment]
Probably use `kubectl port-forward` ? You could also deploy an ingress solution, for a more permanent service.

**Comment by wackazong** [comment]
The way I am doing it now is that I use `socat` to forward the requests to the network. I tried with `iptables` but could not get it to work. Now I just run `socat TCP-LISTEN:80,fork TCP:<LOAD_BALANCER_IP>:80` and now I can connect to the cluster via the load balancer from the host using an http call to localhost. This is what I needed. One of my kubernetes nodes has port 80 forwarding enabled, this is also where the `socat` process is listening. Single point of failure, which does not matter for my dev setup. 

I am pretty confident this is not the most performant solution, if anybody wants to weigh in with an `iptables` solution I am all ears. I found this [example](https://www.mollymiller.net/blog/2020/03/30/0/), which seems to be pretty close to what I want to achieve, yet it did not work.
